packages/ens-utils/src/ethregistrar.ts

```diff
@@ -3,323 +3,351 @@ import {
-);
-export const UNWRAPPED_MAINNET_ETH_REGISTRAR_CONTRACT = buildContractRef(
-  MAINNET,
-  buildAddress("0x57f1887a8BF19b14fC0dF6Fd9B2acc9Af147eA85"),
-);
+  public getManagedSubname = (name: ENSName): ENSName | null => {
+    // must have exactly 2 labels to be a direct subname of ".eth"
+    if (name.labels.length !== 2) return null;
 
-export class NameWrapper implements Registrar {
-  constructor(public contract: ContractRef) {
-    this.contract = contract;
-  }
+    // last label must be "eth"
+    if (name.labels[1] !== ETH_TLD) return null;
 
-  getTokenId(name: ENSName, isWrapped: boolean): TokenId {
-    if (!this.isClaimable(name, isWrapped)) {
-      throw new Error(
-        `Wrapped tokenId for name: "${name.name}" is not claimable by registrar: ${this.contract.address} on chainId: ${this.contract.chain.chainId}`,
+    // NOTE: now we know we have a direct subname of ".eth"
+
+    // first label must be of sufficient length
+    const subnameLength = charCount(name.labels[0]);
+    if (subnameLength < MIN_ETH_REGISTRABLE_LABEL_LENGTH) return null;
+
+    // TODO: also add a check for a maximum length limit as enforced by max block size, etc?
+
+    return buildENSName(name.labels[0]);
+  };
+
+  public getValidatedSubname = (name: ENSName): ENSName => {
+    const subname = this.getManagedSubname(name);
+    if (subname === null)
+      throw new RegistrarUnsupportedNameError(
+        "Name is not directly managed by the .ETH registrar",
+        name,
       );
-    }
-    return buildTokenId(BigInt(namehash(name.name)));
+
+    return subname;
+  };
+
+  public getOnchainRegistrar(): ContractRef {
+    return this.registrarContract;
   }
 
-  isClaimable(name: ENSName, isWrapped: boolean): boolean {
-    if (!isWrapped) return false;
+  public canRegister(
+    name: ENSName,
+    atTimestamp: Timestamp,
+    duration: Duration,
+    existingRegistration?: Registration,
+  ): boolean {
+    if (!this.getManagedSubname(name)) {
+      // name is not directly managed by this registrar
+      return false;
+    }
+
+    if (existingRegistration) {
+      const releaseTimestamp = getDomainReleaseTimestamp(existingRegistration);
 
-    if (name.labels.length >= 2) {
-      if (name.labels[1] === ETH_TLD) {
-        // first label must be of sufficient length
-        if (charCount(name.labels[0]) < MIN_ETH_REGISTRABLE_LABEL_LENGTH)
-          return false;
+      if (releaseTimestamp && releaseTimestamp.time > atTimestamp.time) {
+        // if the name is not yet released, we can't register it
+        // TODO: check for possible off-by-1 errors in the logic above
+        return false;
       }
     }
 
-    // TODO: refine this. For example, there's a maximum length limit, etc.
-    return true;
-  }
-}
+    if (duration.seconds < 1n) {
+      // TODO: enforce that `duration` is the minimum duration allowed for a registration
+      // TODO: need to put the right constant here.
```

Comment:
Suggest to use the old `DEFAULT_REGISTRATION_YEARS` multiplied by `SECONDS_PER_DAY`

Reply:
No it's not a good interface to use years here. It should be defined in seconds as thats how ENS smart contracts work.

The goal described above is totally different though. Please see ENS contracts there's a minimum registration time. You can't register for 1 second.

Reply:
I see, they validate duration is bigger than 0 with below requirement:

<img width="431" alt="Captura de Tela 2024-08-06 às 20 38 09" src="https://github.com/user-attachments/assets/e07614f2-bc8c-4e7e-983f-d1db09224970">

Reply:
Added one suggestion above

Reply:
@FrancoAguzzi I've added the correct constants now for `MIN_REGISTRATION_PERIOD`